syntactic-paths/src/main/java/com/palantir/util/syntacticpath/Path.java

```diff
@@ -330,13 +322,52 @@ private String toStringInternal() {
             String prefix = isAbsolute ? SEPARATOR : "";
             String suffix = isFolder ? SEPARATOR : "";
 
-            return prefix + PATH_JOINER.join(segments) + suffix;
+            return prefix + join(segments) + suffix;
         }
     }
 
     /** Returns the string representation of this (non-normalized) path. */
     @Override
     public String toString() {
-        return stringRepresentation.get();
+        return stringRepresentation;
+    }
+
+    static String join(List<String> segments) {
+        if (segments.size() == 0) {
+            return "";
+        }
+
+        StringBuilder builder = new StringBuilder();
+        Iterator<String> segmentIterator = segments.iterator();
+        builder.append(segmentIterator.next());
+
+        while (segmentIterator.hasNext()) {
+            builder.append("/");
+            builder.append(segmentIterator.next());
+        }
+        return builder.toString();
+    }
+
+    private static List<String> split(String path) {
```

Comment:
would prefer these to be parametrized like this: `splitOmittingEmptyString(String path, String separator)`